server/server.go

```diff
@@ -79,13 +79,30 @@ func (s *IAVLServer) Get(_ context.Context, req *pb.GetRequest) (*pb.GetResponse
 
 	idx, value := s.tree.Get(req.Key)
 	if value == nil {
-		s := status.New(codes.NotFound, "the key requested does not exist")
-		return nil, s.Err()
+		e := status.New(codes.NotFound, "the key requested does not exist")
+		return &pb.GetResponse{Index: idx, Value: nil}, e.Err()
```

Comment:
I don't believe Protobuf differentiates between `[]byte{}` and `nil`, so there's no way for a client to determine whether the key was missing or empty. There's [experimental support](https://github.com/protocolbuffers/protobuf/blob/v3.12.0/docs/field_presence.md) for field presence starting with Protobuf 3.12, but we may not want to rely on that just yet.

Maybe we just add a boolean `missing` field or something for non-existing keys?

Reply:
I was letting the error serve as that indicator. Was I wrong to assume that the error code/message would be transmitted across gRPC along with the protobuf response? 

If so I can add in a missing field, and set the error to nil in both cases.

Reply:
Oh, hm. I don't really know how gRPC handles error codes and response messages, or how the Go implementation does it. Worth trying it out (i.e. writing a test for it).

Reply:
Okay, after some investigation, it is sending the error codes/message over the wire. But using that on the other side is a lot messier than using a value in the protobuf response. So I agree that it makes more sense to add a `Missing` field.